src/Components/TodoList/TodoListItem.jsx

```diff
@@ -0,0 +1,99 @@
+import CheckBox from "../Button/CheckBox";
+import Button from "../Button/Button";
+import styles from "./TodoListItem.module.css";
+import { useContext, useState, useEffect } from "react";
+import { ThemeContext } from "../../Contexts/ThemeContext";
+import { useMediaQuery } from "react-responsive";
+import { AnimatePresence, motion } from "framer-motion";
+
+const TodoListItem = ({
+  id,
+  title,
+  isComplete,
+  onClickCheckBox,
+  onClickEdit,
+  onClickDelete,
+}) => {
+  const { theme } = useContext(ThemeContext);
+  const isMobile = useMediaQuery({ query: "(max-width: 430px)" });
+  const isTablet = useMediaQuery({ query: "(max-width: 768px)" });
+  const textDecoration = isComplete ? "line-through" : "none";
+  const textBoxClass =
+    theme === "light" ? "concave-light-sm" : "concave-dark-md";
+  const handleEdit = () => {
+    onClickEdit(id);
+  };
+
+  const [showButtons, setShowButtons] = useState(isMobile ? false : true);
+  const variants = {
+    open: { x: 0, width: "70%" },
+    closed: { x: 0, width: "100%" },
+    desktop: { x: 0, width: 638 },
+  };
+  const handleDelete = () => {
+    onClickDelete(id);
+  };
+
+  const handleClickCheckBox = (id) => {
+    onClickCheckBox(id);
+  };
+
+  const toggleShowButtonsOnMobile = () => {
+    if (!isMobile) return;
+    setShowButtons(!showButtons);
+  };
+
+  useEffect(() => {
+    setShowButtons(!isMobile);
+  }, [isMobile]);
+
+  return (
+    <div
+      className={styles.wrapper}
+      style={{ gap: `${isTablet ? (showButtons ? "10px" : "0px") : "23px"}` }}
+    >
+      <AnimatePresence>
+        <motion.div
+          animate={isMobile ? (showButtons ? "open" : "closed") : "desktop"}
+          variants={variants}
+          transition={{ type: "ease" }}
+          className={`${styles.textBox} ${textBoxClass} ${styles[showButtons]}`}
+          // key="vweoub"
+        >
+          <CheckBox
+            onClick={handleClickCheckBox}
+            checked={isComplete}
+            itemId={id}
+            keys="vbouwe"
+          />
```

Comment:
위에서 CheckBox 컴포넌트가 ItemId를 받지 않아도 이런식으로 해결할 수 있겠네요.
```suggestion
          <CheckBox
            onClick={() => handleClickCheckBox(id)}
            checked={isComplete}
            keys="vbouwe"
          />
```

Reply:
아 위에 길게 댓글을 달았었는데, 이렇게 쓰면 될 것 같네요!